cardano-db-sync/src/Cardano/DbSync/Types.hs

```diff
@@ -141,14 +144,13 @@ data OffChainPoolResult
   | OffChainPoolResultError !OffChainPoolFetchError
 
 data OffChainVoteResult
-  = OffChainVoteResultMetadata !OffChainVoteData
+  = OffChainVoteResultMetadata
+      !OffChainVoteData
+      (DB.OffChainVoteDataId -> [OffChainVoteAuthor])
+      (DB.OffChainVoteDataId -> [OffChainVoteReference])
+      (DB.OffChainVoteDataId -> [OffChainVoteExternalUpdate])
```

Comment:
To help with naming and readability could this not be something like:
```
data OffChainVoteResult
  = OffChainVoteResultMetadata !OffChainVoteData OffChainVoteAccessors
  | OffChainVoteResultError !OffChainVoteFetchError
 
 dataOffChainVoteAccessors = OffChainVoteAccessors
  { offChainVoteAuthors :: DB.OffChainVoteDataId -> [OffChainVoteAuthor]
  , offChainVoteReferences :: DB.OffChainVoteDataId -> [OffChainVoteReference]
  , offChainVoteExternalUpdates :: DB.OffChainVoteDataId -> [OffChainVoteExternalUpdate]
  }
  ```
  The call site would look something like:
  ```
  insert = \case
  OffChainVoteResultMetadata md accessors -> do
    mocvdId <- DB.insertOffChainVoteData md
    whenJust mocvdId $ \ocvdId -> do
      DB.insertOffChainVoteAuthors $ offChainVoteAuthors accessors ocvdId
      DB.insertOffChainVoteReference $ offChainVoteReferences accessors ocvdId
      DB.insertOffChainVoteExternalUpdate $ offChainVoteExternalUpdates accessors ocvdId
   ```